app/templates/settings/users/index.hbs

```diff
@@ -90,3 +144,179 @@
     </div>
   </div>
 </div>
+
+<AuModal
+  @modalOpen={{this.showBlockMembership}}
+  @closeModal={{this.toggleShowBlockMembership}}
+>
+  <:title>
+    {{t "block-membership"}}
+  </:title>
+  <:body>
+    <div class="au-o-flow au-o-flow--small">
+      <p>{{t "block-membership-explanation"}}</p>
+      <AuFieldset as |f|>
+        <f.legend>{{t "block-membership-which-one"}}</f.legend>
+        <f.content>
+          {{#each this.userBeingBlocked.memberships as |membership|}}
+            {{#let (concat membership.organization.name " - " membership.role.label) as |membershipLabel|}}
+              <AuControlCheckbox
+                @label={{membershipLabel}}
+                @identifier={{membershipLabel}}
+                @onChange={{fn this.toggleMembershipBeingBlocked membership}}
+              />
+            {{/let}}
+          {{/each}}
+        </f.content>
+      </AuFieldset>
+    </div>
+  </:body>
+  <:footer>
+    <AuToolbar
+      as |Group|
+    >
+      <Group>
+        <AuButton
+          @skin="naked"
+          {{on "click" this.toggleShowBlockMembership}}
+        >
+          {{t "cancel"}}
+        </AuButton>
+      </Group>
+      <Group>
+        <AuButton
+          @alert={{true}}
+          @disabled={{is-empty this.membershipsBeingBlocked}}
+          {{on "click" (queue this.blockMemberships this.toggleShowBlockMembership)}}
+        >
+          {{t "block-membership"}}
+        </AuButton>
+      </Group>
+    </AuToolbar>
+  </:footer>
+</AuModal>
```

Comment:
For maintainability/readability maybe better to extract this modal to a separate component?  Same for the unblock memberships modal.